libraries/rush-lib/src/logic/ProjectChangeAnalyzer.ts

```diff
@@ -295,10 +296,12 @@ export class ProjectChangeAnalyzer {
       const lookupByPath: IReadonlyLookupByPath<RushConfigurationProject> =
         this._rushConfiguration.getProjectLookupForRoot(rootDirectory);
 
+      const filterPath: string[] = Array.from(projectSelection ?? []).map((project) => project.projectFolder);
```

Comment:
At minimum the choice to perform filtering needs to be behind a flag in `experiments.json`, because it _will_ break things for consumers with custom plugins, and needs to be a choice whether or not to apply such logic.

Reply:
I added a configuration to determine whether to enable this feature.

Reply:
Hi @dmichon-msft, I've addressed the comments and made the updates. When you get a chance, could you please take another look? Thanks!